api/rest/restapi.go

```diff
@@ -126,12 +126,15 @@ func NewAPIWithHost(ctx context.Context, cfg *Config, h host.Host) (*API, error)
 		ReadHeaderTimeout: cfg.ReadHeaderTimeout,
 		WriteTimeout:      cfg.WriteTimeout,
 		IdleTimeout:       cfg.IdleTimeout,
-		Handler: &ochttp.Handler{
+		Handler:           router,
+	}
+	if cfg.Tracing {
```

Comment:
same, we wrap the handlers in other handlers before creating the server, let's do this a couple of lines before.